usaspending_api/database_scripts/job_archive/backfill_per_transaction_exec_comp.py

```diff
@@ -0,0 +1,322 @@
+
+Expected CLI:
+
+    $ python usaspending_api/database_scripts/job_archive/backfill_per_transaction_exec_comp.py
+
+Purpose:
+
+    Fetch executive compensation data from Broker tables:
+        - published_award_financial_assistance
+        - detached_award_procurement
+    And populate both FABS and FPDS transaction tables
+"""
+import math
+import psycopg2
+import time
+
+from os import environ
+
+
+# DEFINE THESE ENVIRONMENT VARIABLES BEFORE RUNNING!
+SPENDING_CONNECTION_STRING = environ['DATABASE_URL']
+BROKER_CONNECTION_STRING = environ['DATA_BROKER_DATABASE_URL']
+
+
+BROKER_FABS_SELECT_SQL = """
+select
+    published_award_financial_assistance_id,
+    high_comp_officer1_full_na,
+    high_comp_officer1_amount,
+    high_comp_officer2_full_na,
+    high_comp_officer2_amount,
+    high_comp_officer3_full_na,
+    high_comp_officer3_amount,
+    high_comp_officer4_full_na,
+    high_comp_officer4_amount,
+    high_comp_officer5_full_na,
+    high_comp_officer5_amount
+from
+    published_award_financial_assistance
+where
+    published_award_financial_assistance_id between %(min_id)s and %(max_id)s and
+    (
+        high_comp_officer1_amount is not null or
+        high_comp_officer1_full_na is not null or
+        high_comp_officer2_amount is not null or
+        high_comp_officer2_full_na is not null or
+        high_comp_officer3_amount is not null or
+        high_comp_officer3_full_na is not null or
+        high_comp_officer4_amount is not null or
+        high_comp_officer4_full_na is not null or
+        high_comp_officer5_amount is not null or
+        high_comp_officer5_full_na is not null
+    )
+"""
+
+BROKER_FPDS_SELECT_SQL = """
+select
+    detached_award_procurement_id,
+    high_comp_officer1_full_na,
+    high_comp_officer1_amount,
+    high_comp_officer2_full_na,
+    high_comp_officer2_amount,
+    high_comp_officer3_full_na,
+    high_comp_officer3_amount,
+    high_comp_officer4_full_na,
+    high_comp_officer4_amount,
+    high_comp_officer5_full_na,
+    high_comp_officer5_amount
+from
+    detached_award_procurement
+where
+    detached_award_procurement_id between %(min_id)s and %(max_id)s and
+    (
+        high_comp_officer1_amount is not null or
+        high_comp_officer1_full_na is not null or
+        high_comp_officer2_amount is not null or
+        high_comp_officer2_full_na is not null or
+        high_comp_officer3_amount is not null or
+        high_comp_officer3_full_na is not null or
+        high_comp_officer4_amount is not null or
+        high_comp_officer4_full_na is not null or
+        high_comp_officer5_amount is not null or
+        high_comp_officer5_full_na is not null
+    )
+"""
+
+SPENDING_FABS_UPDATE_SQL = """
+update
+    transaction_fabs as fabs set
+        officer_1_name = broker.high_comp_officer1_full_na,
+        officer_1_amount = cast(broker.high_comp_officer1_amount as double precision),
```

Comment:
So these `double precision`s are technically incorrect.  They probably won't cause any problems since all of our numbers have only two decimal places, but you should get in the habit of using `numeric` when dealing with financial values.

https://www.postgresql.org/docs/10/datatype-numeric.html

- numeric - user-specified precision, **exact**
- double precision - variable-precision, **inexact**

Reply:
I will make this change in my next commit along with the comment Brian mentioned above.

Reply:
If a budget item were large enough, it might make double format imprecise even to the second decimal point. We might want to look at this more broadly for things like aggregations where our multi-trillion dollar budget might actually cause problems.

Reply:
We have standardized on `numeric` for the stored values and the aggregations. What did you want to look into?

Reply:
Mostly if we have standardized on numeric for the stored values and the aggregations. I know I haven't been checking so far.

Reply:
I looked at a couple of our DB tables, and it looks like everything is numeric.